pyprep/prep_pipeline.py

```diff
@@ -54,16 +54,38 @@ class PrepPipeline:
     def __init__(self, raw, prep_params, montage, ransac=True, random_state=None):
         """Initialize PREP class."""
         self.raw = raw.copy()
-        self.ch_names = self.raw.ch_names
+
+        # separe eeg and non eeg channels
+        self.raw_non_eeg = self.raw.copy()
+        self.chs_all = self.raw.ch_names.copy()
+        self.chs_types = self.raw.get_channel_types()
+        self.chs_eeg = [
+            self.chs_all[i]
+            for i in range(len(self.chs_all))
+            if self.chs_types[i] == "eeg"
+        ]
+        self.chs_non_eeg = set(self.chs_all) - set(self.chs_eeg)
+        self.raw.pick_channels(self.chs_eeg)
+        self.raw_non_eeg.pick_channels(self.chs_non_eeg)
+
         self.raw.set_montage(montage)
-        self.raw.pick_types(eeg=True, eog=False, meg=False)
-        self.ch_names_eeg = self.raw.ch_names
+
         self.EEG_raw = self.raw.get_data() * 1e6
         self.prep_params = prep_params
+        if self.prep_params["ref_chs"] == "eeg":
+            self.prep_params["ref_chs"] = self.chs_eeg
+        if self.prep_params["reref_chs"] == "eeg":
+            self.prep_params["reref_chs"] = self.chs_eeg
         self.sfreq = self.raw.info["sfreq"]
         self.ransac = ransac
         self.random_state = check_random_state(random_state)
 
+    @property
+    def full_raw(self):
+        """Return a version of self.raw that includes the non-eeg channels."""
+        full_raw = self.raw.copy()
+        return full_raw.add_channels([self.raw_non_eeg])
```

Comment:
I think we should change the way we refer to the different `raw` objects.

IMHO we should have:
1. the raw object with only EEG: `raw_eeg`
1. with all non-eeg: `raw_non_eeg`
1. the full raw, which is here called `full_raw` ... I suggest we call this just `raw`

furthermore, this attribute needs to be documented (like all other publicly accessible attributes actually) in the Docstr of the class.

For an example, see here: https://github.com/mne-tools/mne-python/blob/8fd3c1096c07c1a6dfaa7bb2a4021ba97f02b4fb/mne/io/brainvision/brainvision.py#L54-L57

Reply:
lol, I read this after making my previous comments on the previous discussions. The way you are suggesting is consistent with what we currently have so I will make the change. But do take notice of the cognitive dissonance of 'raw' and a processed signal.